tests/virtualization/xen/ssh_hypervisor.pm

```diff
@@ -26,25 +26,38 @@ sub run {
     my ($self) = @_;
     select_console 'root-console';
     opensusebasetest::select_serial_terminal();
-    my $hypervisor = get_required_var('QAM_XEN_HYPERVISOR');
+    my $hypervisor = get_required_var('HYPERVISOR');
 
-    # Exchange SSH keys
+    # Generate the key pair
     assert_script_run "ssh-keygen -t rsa -P '' -C 'localhost' -f ~/.ssh/id_rsa";
-    assert_script_run "ssh-keyscan $hypervisor > ~/.ssh/known_hosts";
-    exec_and_insert_password "ssh-copy-id root\@$hypervisor";
 
-    # Do the same for normal user because of GUI software
-    assert_script_run "sudo -u $testapi::username sh -c \"ssh-keygen -t rsa -P '' -C 'localhost' -f /home/$testapi::username/.ssh/id_rsa\"";
-    assert_script_run "sudo -u $testapi::username sh -c \"ssh-keyscan $hypervisor > /home/$testapi::username/.ssh/known_hosts\"";
-    exec_and_insert_password "sudo -u $testapi::username sh -c \"ssh-copy-id root\@$hypervisor\"";
+    # Configure the Master socket
+    assert_script_run "echo 'ControlMaster auto
+    ControlPath ~/.ssh/ssh_%r_%h_%p
+    ControlPersist 86400
+    
+    Host javor.qam.suse.cz
+      Hostname javor.qam.suse.cz
+      User root
+        
+    Host sles*
+      ProxyJump javor.qam.suse.cz
+    ' > ~/.ssh/config";
+
```

Comment:
Would you please explain what does the above 'configure master socket' do? And how it helps fluent network connection?

Reply:
Would you please help confirm whether current network scheme can work on osd, especially ssh from qemu worker to guest name?

Reply:
Yes, of course :smile: 
Master socket keeps a session open even after the `ssh` command finish. Then next time the socket can be reused and there's no need to establish new connection. Also there can be more sessions in parallel inside one connection.
The ProxyJump specifies another machine which will be used as a hop between source and destination machines - in our case every connection to `sles*` alias will go through the hypervisor - I should use `$hypervisor` instead of `javor.qam.suse.cz`..

Reply:
To answer your second question: The only thing this schema needs is an SSH connection from the QEMU client to the physical hypervisor. Everything else is encapsulated inside the SSH connection.

Reply:
So this is the core NAT special setting for routing from outside to guests.